src/library/basetrackcache.cpp

```diff
@@ -145,18 +146,59 @@ void BaseTrackCache::setSearchColumns(const QStringList& columns) {
 }
 
 TrackPointer BaseTrackCache::lookupCachedTrack(TrackId trackId) const {
-    TrackPointer pTrack;
-    if (m_bIsCaching) {
-        pTrack = TrackCache::instance().lookupById(trackId).getTrack();
-        // After obtaining a strong reference of the Track object the lock
-        // on TrackCache has been released instantly to reduce lock contention!
-        if (pTrack && pTrack->isDirty()) {
-            m_dirtyTracks.insert(trackId);
+    DEBUG_ASSERT(m_bIsCaching);
+    if (m_cachedTrackId != trackId) {
+        refreshCachedTrack(std::move(trackId));
+    }
+    return m_cachedTrackPtr;
+}
+
+void BaseTrackCache::refreshCachedTrack(TrackId trackId) const {
```

Comment:
This function looks the track up in the GlobalTrackCache and stores it as the local single track.
The overload with the same name, updates only the local single cache. Both functions should be named differently. 
This can be lookupCachedTrack()
We may call the m_cachedTrackPtr -> m_recentTrackPtr, and change the function name accordingly: setRecentTrack(); 
Any other two names works also for me.

Reply:
I didn't notice the naming ambiguity. Now we have separate _refresh_ (from GlobalTrackCache) and _replace_ (with provided TrackPointer) functions.